src/components/sections/contest/ContestDetails/ContestLeaderboard.tsx

```diff
@@ -11,15 +11,28 @@ import {
   TableHeader,
   TableRow,
 } from '@/components/ui/Table'
-import {useGetContestLeaderboard} from '@/lib/queries/contest/getContestLeaderboard'
+import {
+  useGetContestLeaderboard,
+  useGetContestLeaderboardCount,
+} from '@/lib/queries/contest/getContestLeaderboard'
 import {formatAda} from '@/lib/utils/common/format'
+import {useSearchParamsNumericState} from '@/lib/hooks/useSearchParamsState'
+import TablePagination from '@/components/ui/TablePagination'
 
 type ContestLeaderboardProps = {
   contestId: string
 }
 
+export const CONTEST_LEADERBOARD_PAGE_SIZE = 7
```

Comment:
We might even use larger page size here, maybe 10 - 20?

Reply:
Sure, will use 10